Audio loss on rewind and playback crash

### Description

After updating to version 0.6.3, rewinding a video causes the audio to drop out completely until the audio track is manually reselected. If I rewind and fast-forward a few times consecutively, playback crashes entirely.

### Media info

Video:
Title: 1080p HEVC SDR
Codec: HEVC
AVC: No
Profile: Main 10
Level: 120
Resolution: 1920x804
Aspect ratio: 2.40:1
Anamorphic: No
Interlaced: No
Framerate: 23.976025
Bitrate: 5184 kbps
Bit depth: 10 bit
Video range: SDR
Video range type: SDR
Pixel format: yuv420p10le
Ref frames: 1


Audio:
Title: Surround 5.1 - English - Dolby Digital+
Language: eng
Codec: EAC3
AVC: No
Layout: 5.1
Channels: 6 ch
Bitrate: 640 kbps
Sample rate: 48000 Hz
Default: No
Forced: No
External: No

### Player backend

ExoPlayer (default)

### App Version

v0.6.3

### Server Version

10.11.7

### Device

Fire TV Stick Lite

### Logs

```bash
Send App Logs
clientInfo=ClientInfo(name=Wholphin, version=0.6.3-0-ga1595262)
deviceInfo=DeviceInfo(id=7e92b1a5f4c3d0b2, name=FireTVStick)
manufacturer=Amazon
model=AFTSS
apiLevel=28
--------- beginning of main
04-28 04:21:19.119  9307  9376 I chatty  : uid=10107(com.github.damontecres.wholphin) AmazonKeyEventL expire 5 lines
04-28 04:21:23.266  9307 20700 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 2 lines
04-28 04:21:23.341  9307 19531 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 5 lines
04-28 04:21:23.371  9307 21367 I chatty  : uid=10107(com.github.damontecres.wholphin) DefaultDispatch expire 2 lines
04-28 04:21:27.210  9307 21376 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 2 lines
04-28 04:21:27.470  9307 21375 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 4 lines
04-28 04:21:27.470  9307 21370 I chatty  : uid=10107(com.github.damontecres.wholphin) DefaultDispatch expire 2 lines
04-28 04:21:30.645  9307 21369 I chatty  : uid=10107(com.github.damontecres.wholphin) DefaultDispatch expire 7 lines
04-28 04:21:31.333  9307 21383 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:31.397  9307  9312 I chatty  : uid=10107(com.github.damontecres.wholphin) Jit thread pool expire 1 line
04-28 04:21:32.303  9307 21387 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:32.315  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 3 lines
04-28 04:21:32.316  9307 21386 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 6 lines
04-28 04:21:32.321  9307 21387 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 3 lines
04-28 04:21:32.327  9307 21379 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 2 lines
04-28 04:21:32.448  9307 21387 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:32.495  9307 21386 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 6 lines
04-28 04:21:32.512  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 5 lines
04-28 04:21:32.602  9307 21379 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 2 lines
04-28 04:21:32.619  9307 21375 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:32.638  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 2 lines
04-28 04:21:32.650  9307 21401 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:32.693  9307  9331 I chatty  : uid=10107(com.github.damontecres.wholphin) RenderThread expire 2 lines
04-28 04:21:32.694  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:32.703  9307 21379 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 4 lines
04-28 04:21:32.711  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 4 lines
04-28 04:21:32.730  9307 21149 I chatty  : uid=10107(com.github.damontecres.wholphin) DefaultDispatch expire 2 lines
04-28 04:21:32.756  9307 21369 I chatty  : uid=10107(com.github.damontecres.wholphin) DefaultDispatch expire 1 line
04-28 04:21:32.867  9307 20700 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:32.875  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:32.896  9307 21400 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:33.069  9307  9317 I chatty  : uid=10107(com.github.damontecres.wholphin) HeapTaskDaemon expire 3 lines
04-28 04:21:33.686  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 3 lines
04-28 04:21:34.326  9307  9321 I chatty  : uid=10107(com.github.damontecres.wholphin) Profile Saver expire 2 lines
04-28 04:21:34.349  9307  9317 I chatty  : uid=10107(com.github.damontecres.wholphin) HeapTaskDaemon expire 2 lines
04-28 04:21:35.638  9307 21407 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:35.645  9307 21406 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 6 lines
04-28 04:21:35.649  9307 21407 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 3 lines
04-28 04:21:35.651  9307 21379 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:35.715  9307 21407 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:35.768  9307 21406 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 7 lines
04-28 04:21:35.798  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:35.901  9307 21369 I chatty  : uid=10107(com.github.damontecres.wholphin) DefaultDispatch expire 1 line
04-28 04:21:35.928  9307 21416 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:36.115  9307 21404 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 8 lines
04-28 04:21:36.454  9307  9317 I chatty  : uid=10107(com.github.damontecres.wholphin) HeapTaskDaemon expire 2 lines
04-28 04:21:37.000  9307 21404 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 10 lines
04-28 04:21:37.503  9307  9317 I chatty  : uid=10107(com.github.damontecres.wholphin) HeapTaskDaemon expire 1 line
04-28 04:21:37.686  9307 21369 I chatty  : uid=10107(com.github.damontecres.wholphin) DefaultDispatch expire 1 line
04-28 04:21:38.116  9307  9317 I chatty  : uid=10107(com.github.damontecres.wholphin) HeapTaskDaemon expire 1 line
04-28 04:21:38.127  9307 21404 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 12 lines
04-28 04:21:38.529  9307  9317 I chatty  : uid=10107(com.github.damontecres.wholphin) HeapTaskDaemon expire 2 lines
04-28 04:21:39.030  9307 21404 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 9 lines
04-28 04:21:39.384  9307  9317 I chatty  : uid=10107(com.github.damontecres.wholphin) HeapTaskDaemon expire 1 line
04-28 04:21:40.704  9307 21406 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:40.927  9307 21376 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 2 lines
04-28 04:21:43.133  9307 21404 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 3 lines
04-28 04:21:45.698  9307  9317 I chatty  : uid=10107(com.github.damontecres.wholphin) HeapTaskDaemon expire 1 line
04-28 04:21:45.757  9307 21406 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:45.982  9307 21369 I chatty  : uid=10107(com.github.damontecres.wholphin) DefaultDispatch expire 1 line
04-28 04:21:47.897  9307 21376 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 2 lines
04-28 04:21:47.906  9307 21379 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:47.913  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 7 lines
04-28 04:21:48.007  9307 21418 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:48.307  9307 21379 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 12 lines
04-28 04:21:48.432  9307 21421 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 2 lines
04-28 04:21:48.404  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:48.443  9307 21420 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 5 lines
04-28 04:21:48.458  9307 21379 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:48.475  9307 21421 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 3 lines
04-28 04:21:48.542  9307 21420 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 4 lines
04-28 04:21:48.836  9307 21379 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:49.022  9307 21420 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 4 lines
04-28 04:21:49.034  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:49.117  9307 21433 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:49.127  9307 21367 I chatty  : uid=10107(com.github.damontecres.wholphin) DefaultDispatch expire 1 line
04-28 04:21:49.151  9307  9317 I chatty  : uid=10107(com.github.damontecres.wholphin) HeapTaskDaemon expire 6 lines
04-28 04:21:53.543  9307 21420 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:54.171  9307 21367 I chatty  : uid=10107(com.github.damontecres.wholphin) DefaultDispatch expire 2 lines
04-28 04:21:57.150  9307 21379 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 2 lines
04-28 04:21:57.184  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:57.195  9307 21434 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:57.204  9307 21420 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 3 lines
04-28 04:21:57.255  9307 12378 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:57.286  9307 21379 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 2 lines
04-28 04:21:57.319  9307 12378 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:57.324  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:57.334  9307 21437 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:57.338  9307 21420 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 3 lines
04-28 04:21:57.498  9307 21367 I chatty  : uid=10107(com.github.damontecres.wholphin) DefaultDispatch expire 1 line
04-28 04:21:57.739  9307  9317 I chatty  : uid=10107(com.github.damontecres.wholphin) HeapTaskDaemon expire 1 line
04-28 04:21:58.016  9307 21379 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:58.054  9307  9307 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:58.034  9307 21375 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:58.065  9307 21440 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 1 line
04-28 04:21:58.200  9307 21379 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 2 lines
04-28 04:21:58.325  9307  9317 I chatty  : uid=10107(com.github.damontecres.wholphin) HeapTaskDaemon expire 1 line
04-28 04:21:58.379  9307 21420 I chatty  : uid=10107(com.github.damontecres.wholphin) expire 4 lines
04-28 04:21:58.677  9307 21367 I chatty  : uid=10107(com.github.damontecres.wholphin) DefaultDispatch expire 2 lines
04-28 04:21:59.759  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 17270(2MB) AllocSpace objects, 67(4MB) LOS objects, 7% free, 74MB/80MB, paused 171us total 194.678ms
04-28 04:21:59.897  9307 21379 I MediaCodecLogger: Flushing: 9307.HW.video.hevc.framerate = 23.974 fps
04-28 04:21:59.944  9307  9307 W ExoPlayer:Audio: type=1400 audit(0.0:302): avc: denied { read } for name="u:object_r:media_prop:s0" dev="tmpfs" ino=1330 scontext=u:r:untrusted_app:s0:c107,c256,c512,c768 tcontext=u:object_r:media_prop:s0 tclass=file permissive=0
04-28 04:21:59.910  9307 21369 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:21:59.929  9307 21379 I MediaCodecLogger: Flushing: 9307.HW.video.hevc.framerate = 0.000 fps
04-28 04:22:00.148  9307 21420 I MediaCodecLogger: updatePTSTime [HW.video.hevc] First PTS after Flush or reset = 1000147147000
04-28 04:22:00.148  9307 21420 I MediaCodecLogger: updatePTSTime Possible seek found! last pts:1000111987000 cur pts:1000147147000
04-28 04:22:00.573  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 18765(3MB) AllocSpace objects, 33(3MB) LOS objects, 7% free, 74MB/80MB, paused 2.211ms total 304.263ms
04-28 04:22:00.598  9307 21379 I android.media.AudioTrack: AUDIOINFO: audio_input: format: 2 sample_rate: 48000 channels: 6 AudioAttributes: AudioAttributes: usage=USAGE_MEDIA content=CONTENT_TYPE_MOVIE flags=0x200 tags= bundle=null
04-28 04:22:01.388  9307 21420 I MediaCodecLogger: the latest buffer is 1495 old with packageName com.github.damontecres.wholphin. Possible resume detected
04-28 04:22:01.389  9307 21420 I MediaCodecLogger: Rendering first frame after starting or flushing.
04-28 04:22:01.412  9307 21367 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:22:01.450  9307 21444 E FrameEvents: updateAcquireFence: Did not find frame.
04-28 04:22:01.609  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 21611(1157KB) AllocSpace objects, 69(5MB) LOS objects, 7% free, 75MB/81MB, paused 686us total 212.449ms
04-28 04:22:02.682  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 16089(1028KB) AllocSpace objects, 73(6MB) LOS objects, 7% free, 74MB/80MB, paused 169us total 185.330ms
04-28 04:22:03.607  9307 21420 I MediaCodecLogger: 9307.HW.video.hevc.bitrateInKbps = 14803
04-28 04:22:04.169  9307 21367 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:22:08.696  9307 21420 I MediaCodecLogger: 9307.HW.video.hevc.bitrateInKbps = 3045
04-28 04:22:09.168  9307 21369 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:22:13.438  9307  9307 I Wholphin: Current Destination: SeriesOverview(itemId=3366353b-ebf3-01d0-2d25-5ad0375d7480, type=Series, seasonEpisode=SeasonEpisodeIds(seasonId=720bbe80-511b-9911-3026-18e0d3affef3, seasonNumber=4, episodeId=4cefaa49-49d9-3c18-7a39-2afc247c28c3, episodeNumber=1))
04-28 04:22:13.466  9307 21149 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Stopped
04-28 04:22:13.468  9307 21379 I MediaCodecLogger: Flushing: 9307.HW.video.hevc.framerate = 23.976 fps
04-28 04:22:13.494  9307  9307 W ExoPlayer:Audio: type=1400 audit(0.0:303): avc: denied { read } for name="u:object_r:media_prop:s0" dev="tmpfs" ino=1330 scontext=u:r:untrusted_app:s0:c107,c256,c512,c768 tcontext=u:object_r:media_prop:s0 tclass=file permissive=0
04-28 04:22:13.527  9307 21379 I MediaCodecLogger: 9307.HW.video.hevc.TotalVideoPlaybackTimeMs = 37455
04-28 04:22:13.527  9307 21379 I MediaCodecLogger: 9307.HW.video.hevc.StreamAvgFramerate = 23.975
04-28 04:22:13.527  9307 21379 I MediaCodecLogger: 9307.HW.video.hevc.videoResolution.FHD = 37455
04-28 04:22:13.528  9307 21379 I MediaCodecLogger: 9307.HW.video.hevc.StreamFramerate = 23.292
04-28 04:22:13.528  9307 21379 I MediaCodecLogger: 9307.HW.video.hevc.AspectRatio = 2.388
04-28 04:22:13.528  9307 21379 I MediaCodecLogger: 9307.HW.video.hevc.ASAP-P5Ms = 480
04-28 04:22:13.528  9307 21379 I MediaCodecLogger: 9307.HW.video.hevc.ASAP-P2Ms = 9
04-28 04:22:13.528  9307 21379 I MediaCodecLogger: 9307.HW.video.hevc.ASAP-P3Ms = 19
04-28 04:22:13.528  9307 21379 I MediaCodecLogger: 9307.HW.video.hevc.FirstFrameReadyMs = 55
04-28 04:22:13.528  9307 21379 I MediaCodecLogger: 9307.HW.video.hevc.StreamAvgBitrateInKbps = 9283
04-28 04:22:13.514  9307  9307 W ExoPlayer:Playb: type=1400 audit(0.0:304): avc: denied { read } for name="u:object_r:media_prop:s0" dev="tmpfs" ino=1330 scontext=u:r:untrusted_app:s0:c107,c256,c512,c768 tcontext=u:object_r:media_prop:s0 tclass=file permissive=0
04-28 04:22:13.551  9307  9307 I MediaSessionImpl: Release f7528a [AndroidXMedia3/1.10.0] [sheldon, AFTSS, Amazon, 28] [media3.common, media3.exoplayer, media3.decoder, media3.session, media3.datasource, media3.ui.compose, media3.extractor, media3.decoder.ffmpeg, media3.exoplayer.hls]
04-28 04:22:13.558  9307  9307 I DefaultSocketApi: Subscription count changed to 1
04-28 04:22:13.691  9307  9321 I tecres.wholphi: Waiting for a blocking GC ProfileSaver
04-28 04:22:13.817  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 47664(3MB) AllocSpace objects, 943(62MB) LOS objects, 26% free, 16MB/22MB, paused 2.525ms total 254.590ms
04-28 04:22:13.817  9307  9321 I tecres.wholphi: WaitForGcToComplete blocked ProfileSaver on HeapTrim for 126.555ms
04-28 04:22:14.580  9307 21149 I OkHttpClient: GET http://192.168.1.128:8096/Items/720bbe80-511b-9911-3026-18e0d3affef3/SpecialFeatures
04-28 04:22:14.582  9307  9307 I Wholphin: Current tab: 3, series_overview
04-28 04:22:14.583  9307 21369 I OkHttpClient: GET http://192.168.1.128:8096/Shows/3366353b-ebf3-01d0-2d25-5ad0375d7480/Episodes?fields=MediaSources&fields=MediaSourceCount&fields=Overview&fields=CustomRating&fields=PrimaryImageAspectRatio&fields=CanDelete&seasonId=720bbe80-511b-9911-3026-18e0d3affef3&startIndex=0&limit=100&sortBy=IndexNumber
04-28 04:22:14.629  9307  9307 I Choreographer: Skipped 63 frames!  The application may be doing too much work on its main thread.
04-28 04:22:14.930  9307  9331 I OpenGLRenderer: Davey! duration=1464ms; Flags=0, IntendedVsync=7039211757998, Vsync=7039295174748, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=7039306266111, AnimationStart=7039306272034, PerformTraversalsStart=7039441811573, DrawStart=7039441953188, SyncQueued=7040315884342, SyncStart=7040315939881, IssueDrawCommandsStart=7040318245650, SwapBuffers=7040675609034, FrameCompleted=7040676524881, DequeueBufferDuration=272000, QueueBufferDuration=305000, 
04-28 04:22:14.979  9307 21370 I OkHttpClient: GET http://192.168.1.128:8096/Items/3366353b-ebf3-01d0-2d25-5ad0375d7480/ThemeSongs?inheritFromParent=false
04-28 04:22:14.986  9307  9307 I MediaSessionImpl: Release f7528a [AndroidXMedia3/1.10.0] [sheldon, AFTSS, Amazon, 28] [media3.common, media3.exoplayer, media3.decoder, media3.session, media3.datasource, media3.ui.compose, media3.extractor, media3.decoder.ffmpeg, media3.exoplayer.hls]
04-28 04:22:15.269  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 59850(5MB) AllocSpace objects, 24(1140KB) LOS objects, 28% free, 15MB/21MB, paused 165us total 222.896ms
04-28 04:22:15.274  9307  9315 W System  : A resource failed to call end. 
04-28 04:22:15.274  9307  9315 I chatty  : uid=10107(com.github.damontecres.wholphin) FinalizerDaemon identical 1 line
04-28 04:22:15.274  9307  9315 W System  : A resource failed to call end. 
04-28 04:25:17.015  9307  9307 I Wholphin: Current Destination: Playback(itemId=4cefaa49-49d9-3c18-7a39-2afc247c28c3, positionMs=0, itemPlayback=null, forceTranscoding=false)
04-28 04:25:17.060  9307 21367 I Wholphin: Stopping music
04-28 04:25:17.060  9307 21367 W Wholphin: Stopping but no MediaSession
04-28 04:25:17.168  9307 21367 I OkHttpClient: GET http://192.168.1.128:8096/Items/4cefaa49-49d9-3c18-7a39-2afc247c28c3
04-28 04:25:17.331  9307 21367 I OkHttpClient: GET http://192.168.1.128:8096/Items/4cefaa49-49d9-3c18-7a39-2afc247c28c3/Intros?userId=f82b7c41-93e5-4d10-b28a-7c9d1e2f3a4b
04-28 04:25:17.339  9307 21367 I Wholphin: Playing 4cefaa49-49d9-3c18-7a39-2afc247c28c3
04-28 04:25:17.354  9307 21367 I Wholphin: Switching player backend to EXO_PLAYER
04-28 04:25:17.370  9307 21367 I DefaultRenderersFactory: Loaded FfmpegAudioRenderer.
04-28 04:25:17.384  9307 21367 I MediaSessionImpl: Init bfea2e7 [AndroidXMedia3/1.10.0] [sheldon, AFTSS, Amazon, 28]
04-28 04:25:17.399  9307 21149 I OkHttpClient: POST http://192.168.1.128:8096/Items/4cefaa49-49d9-3c18-7a39-2afc247c28c3/PlaybackInfo
04-28 04:25:17.437  9307 21149 I Wholphin: Playback decision for 4cefaa49-49d9-3c18-7a39-2afc247c28c3: DirectPlay
04-28 04:25:17.451  9307  9307 I AssHandler: onMediaItemTransition: item = androidx.media3.common.MediaItem@c177aed5, reason = 3
04-28 04:25:17.513  9307 21370 I OkHttpClient: GET http://192.168.1.128:8096/Shows/3366353b-ebf3-01d0-2d25-5ad0375d7480/Episodes?fields=PrimaryImageAspectRatio&fields=SeasonUserData&fields=ChildCount&fields=Overview&fields=Trickplay&fields=SortName&fields=Chapters&fields=MediaSources&fields=MediaSourceCount&fields=ParentId&fields=CanDelete&startItemId=4cefaa49-49d9-3c18-7a39-2afc247c28c3&limit=100
04-28 04:25:17.513  9307 21367 I OkHttpClient: GET http://192.168.1.128:8096/MediaSegments/4cefaa49-49d9-3c18-7a39-2afc247c28c3
04-28 04:25:17.572  9307 21571 I AssHandler: setVideoSize: width = 1920, height = 804
04-28 04:25:17.592  9307  9307 I AssHandler: onSurfaceSizeChanged: width = 1920, height = 1080
04-28 04:25:17.915  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 125535(6MB) AllocSpace objects, 13(2MB) LOS objects, 26% free, 16MB/22MB, paused 1.643ms total 287.434ms
04-28 04:25:17.919  9307  9315 W System  : A resource failed to call dispose. 
04-28 04:25:18.005  9307 21573 I OMXClient: IOmx service obtained
04-28 04:25:18.010  9307  9307 I AssHandler: onTracksChanged androidx.media3.common.Tracks@74b8895b
04-28 04:25:18.011  9307  9307 I AssHandler: setVideoSize: width = 1920, height = 804
04-28 04:25:18.011  9307  9307 I AssHandler: subtitle track disabled
04-28 04:25:18.016  9307 21572 I MediaCodecLogger: [ASAP] HW.video.hevc P2 + 7223762
04-28 04:25:18.017  9307 21572 I MediaCodecLogger: HW.video.hevc Adaptive Streaming max w:1920 h:804, screen size:FHD
04-28 04:25:18.022  9307 21573 I ACodec  : codec does not support config priority (err -1010)
04-28 04:25:18.030  9307 21572 I MediaCodecLogger: [ASAP] HW.video.hevc P2 - 7223776
04-28 04:25:18.031  9307 21566 I MediaCodecLogger: [ASAP] HW.video.hevc P3 + 7223777
04-28 04:25:18.052  9307 21573 W ACodec  : [OMX.MTK.VIDEO.DECODER.HEVC] setting nBufferCountActual to 6 failed: -22
04-28 04:25:18.055  9307 21572 I MediaCodecLogger: [ASAP] HW.video.hevc P3 - 7223801
04-28 04:25:18.074  9307 21572 I MediaCodecLogger: [ASAP] HW.video.hevc Got First Input Frame 7223820 pts:1000000000000
04-28 04:25:18.080  9307 21566 I android.media.AudioTrack: AUDIOINFO: audio_input: format: 2 sample_rate: 48000 channels: 6 AudioAttributes: AudioAttributes: usage=USAGE_MEDIA content=CONTENT_TYPE_MOVIE flags=0x200 tags= bundle=null
04-28 04:25:18.140  9307  9307 I AssHandler: onTracksChanged androidx.media3.common.Tracks@74b88955
04-28 04:25:18.140  9307  9307 I AssHandler: setVideoSize: width = 1920, height = 804
04-28 04:25:18.140  9307  9307 I AssHandler: subtitle track disabled
04-28 04:25:18.164  9307 21573 W AHierarchicalStateMachine: Warning message AMessage(what = 'omxI') = {
04-28 04:25:18.164  9307 21573 W AHierarchicalStateMachine:   int32_t type = 0
04-28 04:25:18.164  9307 21573 W AHierarchicalStateMachine:   int32_t event = 3
04-28 04:25:18.164  9307 21573 W AHierarchicalStateMachine:   int32_t data1 = 1
04-28 04:25:18.164  9307 21573 W AHierarchicalStateMachine:   int32_t data2 = 117440527
04-28 04:25:18.164  9307 21573 W AHierarchicalStateMachine: } unhandled in root state.
04-28 04:25:18.175  9307 21573 W ACodec  : [OMX.MTK.VIDEO.DECODER.HEVC] setting nBufferCountActual to 12 failed: -22
04-28 04:25:18.215  9307 21572 I MediaCodecLogger: updateFormatChanged width = 1920 height = 804
04-28 04:25:18.215  9307 21572 I MediaCodecLogger: updatePTSTime [HW.video.hevc] First PTS after Flush or reset = 1000000000000
04-28 04:25:18.215  9307 21572 I MediaCodecLogger: [ASAP] HW.video.hevc Got First Frame Ready 7223962(142)
04-28 04:25:18.225  9307 21572 I MediaCodecLogger: Rendering first frame after starting or flushing.
04-28 04:25:18.225  9307 21572 I MediaCodecLogger: [ASAP] HW.video.hevc Got First Frame Render 7223971(151)
04-28 04:25:18.225  9307 21572 I MediaCodecLogger: App sets Timestamps for render in SurfaceFlinger
04-28 04:25:18.243  9307  9307 I AssHandler: onVideoSizeChanged: width = 1920, height = 804
04-28 04:25:18.264  9307 21584 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing
04-28 04:25:18.287  9307  9307 I AssHandler: onSurfaceSizeChanged: width = 1920, height = 804
04-28 04:25:18.332  9307  9331 E IMGSRV  : :0: IsTextureConsistent: IMGEGLImage is not consistent
04-28 04:25:18.339  9307  9331 E IMGSRV  : :0: IsTextureConsistent: IMGEGLImage is not consistent
04-28 04:25:18.348  9307 21589 E FrameEvents: updateAcquireFence: Did not find frame.
04-28 04:25:18.573  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 94403(4MB) AllocSpace objects, 80(2MB) LOS objects, 20% free, 22MB/28MB, paused 158us total 212.403ms
04-28 04:25:18.732  9307 21572 I MediaCodecLogger: 9307.HW.video.hevc Resolution change from 0x0 to 1920x804 took = 5
04-28 04:25:18.855  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 91696(5MB) AllocSpace objects, 466(9MB) LOS objects, 21% free, 21MB/27MB, paused 168us total 196.280ms
04-28 04:25:19.250  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 38545(2MB) AllocSpace objects, 308(6MB) LOS objects, 20% free, 23MB/29MB, paused 165us total 203.798ms
04-28 04:25:20.037  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 6390(273KB) AllocSpace objects, 0(0B) LOS objects, 15% free, 31MB/37MB, paused 277us total 168.859ms
04-28 04:25:20.566  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 3592(391KB) AllocSpace objects, 53(4MB) LOS objects, 14% free, 36MB/42MB, paused 162us total 176.671ms
04-28 04:25:21.078  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 4397(351KB) AllocSpace objects, 37(3MB) LOS objects, 12% free, 41MB/47MB, paused 182us total 193.440ms
04-28 04:25:21.530  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 2757(304KB) AllocSpace objects, 32(2MB) LOS objects, 11% free, 47MB/53MB, paused 174us total 179.927ms
04-28 04:25:22.007  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 2669(269KB) AllocSpace objects, 33(3MB) LOS objects, 10% free, 51MB/57MB, paused 987us total 192.019ms
04-28 04:25:22.366  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 2195(310KB) AllocSpace objects, 18(1220KB) LOS objects, 9% free, 58MB/64MB, paused 394us total 190.661ms
04-28 04:25:22.763  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 2393(330KB) AllocSpace objects, 25(2MB) LOS objects, 8% free, 61MB/67MB, paused 176us total 183.858ms
04-28 04:25:23.079  9307 21572 I MediaCodecLogger: 9307.HW.video.hevc.bitrateInKbps = 6787
04-28 04:25:23.342  9307 21370 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:25:24.591  9307 21566 I MediaCodecLogger: Flushing: 9307.HW.video.hevc.framerate = 23.975 fps
04-28 04:25:24.604  9307 21370 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:25:24.627  9307 21566 I android.media.AudioTrack: AUDIOINFO: audio_input: format: 2 sample_rate: 48000 channels: 6 AudioAttributes: AudioAttributes: usage=USAGE_MEDIA content=CONTENT_TYPE_MOVIE flags=0x200 tags= bundle=null
04-28 04:25:24.648  9307 21572 I MediaCodecLogger: updatePTSTime [HW.video.hevc] First PTS after Flush or reset = 1000020854000
04-28 04:25:24.634  9307  9307 W ExoPlayer:Audio: type=1400 audit(0.0:306): avc: denied { read } for name="u:object_r:media_prop:s0" dev="tmpfs" ino=1330 scontext=u:r:untrusted_app:s0:c107,c256,c512,c768 tcontext=u:object_r:media_prop:s0 tclass=file permissive=0
04-28 04:25:24.648  9307 21572 I MediaCodecLogger: updatePTSTime Possible seek found! last pts:1000006632000 cur pts:1000020854000
04-28 04:25:24.697  9307 21572 I MediaCodecLogger: updatePTSTime [HW.video.hevc] First PTS after Flush or reset = 1000051301000
04-28 04:25:24.697  9307 21572 I MediaCodecLogger: updatePTSTime Possible seek found! last pts:1000020854000 cur pts:1000051301000
04-28 04:25:24.684  9307  9307 W ExoPlayer:Audio: type=1400 audit(0.0:307): avc: denied { read } for name="u:object_r:media_prop:s0" dev="tmpfs" ino=1330 scontext=u:r:untrusted_app:s0:c107,c256,c512,c768 tcontext=u:object_r:media_prop:s0 tclass=file permissive=0
04-28 04:25:24.705  9307 21566 I android.media.AudioTrack: AUDIOINFO: audio_input: format: 2 sample_rate: 48000 channels: 6 AudioAttributes: AudioAttributes: usage=USAGE_MEDIA content=CONTENT_TYPE_MOVIE flags=0x200 tags= bundle=null
04-28 04:25:24.729  9307 21572 I MediaCodecLogger: Rendering first frame after starting or flushing.
04-28 04:25:24.773  9307 21584 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal: Playback error
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:   androidx.media3.exoplayer.ExoPlaybackException: Unexpected runtime error
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:925)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at android.os.Handler.dispatchMessage(Handler.java:102)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at android.os.Looper.loop(Looper.java:211)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at android.os.HandlerThread.run(HandlerThread.java:65)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:   Caused by: kotlin.UninitializedPropertyAccessException: lateinit property handler has not been initialized
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at io.github.peerless2012.ass.media.widget.AssSubtitleTextureView$AssRenderThread.requestRender(AssSubtitleTextureView.kt:123)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at io.github.peerless2012.ass.media.widget.AssSubtitleTextureView.requestRender(AssSubtitleTextureView.kt:69)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at io.github.peerless2012.ass.media.widget.AssSubtitleView.onAttachedToWindow$lambda$1(AssSubtitleView.kt:58)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at io.github.peerless2012.ass.media.widget.AssSubtitleView$$ExternalSyntheticLambda0.invoke(D8$$SyntheticClass:0)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at io.github.peerless2012.ass.media.AssHandler.setVideoTime(AssHandler.kt:79)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at io.github.peerless2012.ass.media.render.AssRenderer.render(AssRenderer.kt:23)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.RendererHolder.render(RendererHolder.java:392)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:1388)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:711)
04-28 04:25:25.539  9307 21566 E ExoPlayerImplInternal:       ... 3 more
04-28 04:25:25.540  9307 21566 I MediaCodecLogger: Flushing: 9307.HW.video.hevc.framerate = 23.963 fps
04-28 04:25:25.573  9307  9307 I AssHandler: onVideoSizeChanged: width = 0, height = 0
04-28 04:25:25.584  9307  9307 W ExoPlayer:Audio: type=1400 audit(0.0:308): avc: denied { read } for name="u:object_r:media_prop:s0" dev="tmpfs" ino=1330 scontext=u:r:untrusted_app:s0:c107,c256,c512,c768 tcontext=u:object_r:media_prop:s0 tclass=file permissive=0
04-28 04:25:25.626  9307 21566 I MediaCodecLogger: 9307.HW.video.hevc.TotalVideoPlaybackTimeMs = 7717
04-28 04:25:25.626  9307 21566 I MediaCodecLogger: 9307.HW.video.hevc.StreamAvgFramerate = 23.973
04-28 04:25:25.626  9307 21566 I MediaCodecLogger: 9307.HW.video.hevc.videoResolution.FHD = 7717
04-28 04:25:25.626  9307 21566 I MediaCodecLogger: 9307.HW.video.hevc.StreamFramerate = 0.000
04-28 04:25:25.626  9307 21566 I MediaCodecLogger: 9307.HW.video.hevc.AspectRatio = 2.388
04-28 04:25:25.627  9307 21566 I MediaCodecLogger: 9307.HW.video.hevc.ASAP-P5Ms = 151
04-28 04:25:25.627  9307 21566 I MediaCodecLogger: 9307.HW.video.hevc.ASAP-P2Ms = 14
04-28 04:25:25.627  9307 21566 I MediaCodecLogger: 9307.HW.video.hevc.ASAP-P3Ms = 24
04-28 04:25:25.627  9307 21566 I MediaCodecLogger: 9307.HW.video.hevc.FirstFrameReadyMs = 142
04-28 04:25:25.627  9307 21566 I MediaCodecLogger: 9307.HW.video.hevc.StreamAvgBitrateInKbps = 6787
04-28 04:25:25.624  9307  9307 W ExoPlayer:Playb: type=1400 audit(0.0:309): avc: denied { read } for name="u:object_r:media_prop:s0" dev="tmpfs" ino=1330 scontext=u:r:untrusted_app:s0:c107,c256,c512,c768 tcontext=u:object_r:media_prop:s0 tclass=file permissive=0
04-28 04:25:25.647  9307 21149 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:25:25.671  9307  9307 E Wholphin: Playback error
04-28 04:25:25.671  9307  9307 E Wholphin: androidx.media3.exoplayer.ExoPlaybackException: Unexpected runtime error
04-28 04:25:25.671  9307  9307 E Wholphin: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:925)
04-28 04:25:25.671  9307  9307 E Wholphin: 	at android.os.Handler.dispatchMessage(Handler.java:102)
04-28 04:25:25.671  9307  9307 E Wholphin: 	at android.os.Looper.loop(Looper.java:211)
04-28 04:25:25.671  9307  9307 E Wholphin: 	at android.os.HandlerThread.run(HandlerThread.java:65)
04-28 04:25:25.671  9307  9307 E Wholphin: Caused by: kotlin.UninitializedPropertyAccessException: lateinit property handler has not been initialized
04-28 04:25:25.671  9307  9307 E Wholphin: 	at io.github.peerless2012.ass.media.widget.AssSubtitleTextureView$AssRenderThread.requestRender(AssSubtitleTextureView.kt:123)
04-28 04:25:25.671  9307  9307 E Wholphin: 	at io.github.peerless2012.ass.media.widget.AssSubtitleTextureView.requestRender(AssSubtitleTextureView.kt:69)
04-28 04:25:25.671  9307  9307 E Wholphin: 	at io.github.peerless2012.ass.media.widget.AssSubtitleView.onAttachedToWindow$lambda$1(AssSubtitleView.kt:58)
04-28 04:25:25.671  9307  9307 E Wholphin: 	at io.github.peerless2012.ass.media.widget.AssSubtitleView$$ExternalSyntheticLambda0.invoke(D8$$SyntheticClass:0)
04-28 04:25:25.671  9307  9307 E Wholphin: 	at io.github.peerless2012.ass.media.AssHandler.setVideoTime(AssHandler.kt:79)
04-28 04:25:25.671  9307  9307 E Wholphin: 	at io.github.peerless2012.ass.media.render.AssRenderer.render(AssRenderer.kt:23)
04-28 04:25:25.671  9307  9307 E Wholphin: 	at androidx.media3.exoplayer.RendererHolder.render(RendererHolder.java:392)
04-28 04:25:25.671  9307  9307 E Wholphin: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:1388)
04-28 04:25:25.671  9307  9307 E Wholphin: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:711)
04-28 04:25:25.671  9307  9307 E Wholphin: 	... 3 more
04-28 04:25:25.683  9307  9307 W Wholphin: Playback error during DirectPlay, falling back to transcoding
04-28 04:25:25.698  9307 21149 I OkHttpClient: POST http://192.168.1.128:8096/Items/4cefaa49-49d9-3c18-7a39-2afc247c28c3/PlaybackInfo
04-28 04:25:25.770  9307 21620 E Surface : getSlotFromBufferLocked: unknown buffer: 0xa4159740
04-28 04:25:25.770  9307 21620 E IMGSRV  : :0: QueueCancelBufferWrapper: Failed to cancel buffer 0x8dd39448 (-22)
04-28 04:25:25.821  9307 21149 I Wholphin: Playback decision for 4cefaa49-49d9-3c18-7a39-2afc247c28c3: Transcode
04-28 04:25:25.835  9307 21584 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Stopped
04-28 04:25:25.839  9307  9307 I AssHandler: onMediaItemTransition: item = androidx.media3.common.MediaItem@2d3458db, reason = 3
04-28 04:25:25.840  9307  9307 I AssHandler: onTracksChanged androidx.media3.common.Tracks@1
04-28 04:25:25.840  9307  9307 I AssHandler: subtitle track disabled
04-28 04:25:25.891  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 31281(1650KB) AllocSpace objects, 593(39MB) LOS objects, 17% free, 28MB/34MB, paused 181us total 482.936ms
04-28 04:25:26.462  9307  9307 I AssHandler: onTracksChanged androidx.media3.common.Tracks@7d5e877a
04-28 04:25:26.462  9307  9307 I AssHandler: setVideoSize: width = 1920, height = 804
04-28 04:25:26.463  9307  9307 I AssHandler: subtitle track disabled
04-28 04:25:26.579  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 39893(5MB) AllocSpace objects, 242(15MB) LOS objects, 27% free, 15MB/21MB, paused 236us total 277.205ms
04-28 04:25:26.586  9307  9315 W System  : A resource failed to call end. 
04-28 04:25:26.586  9307  9315 I chatty  : uid=10107(com.github.damontecres.wholphin) FinalizerDaemon identical 1 line
04-28 04:25:26.586  9307  9315 W System  : A resource failed to call end. 
04-28 04:25:26.873  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 57278(5MB) AllocSpace objects, 223(4MB) LOS objects, 23% free, 19MB/25MB, paused 154us total 201.293ms
04-28 04:25:27.163  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 61991(4MB) AllocSpace objects, 456(8MB) LOS objects, 26% free, 16MB/22MB, paused 258us total 185.192ms
04-28 04:25:28.418  9307 21641 I OMXClient: IOmx service obtained
04-28 04:25:28.426  9307 21640 I MediaCodecLogger: [ASAP] HW.video.avc P2 + 7234172
04-28 04:25:28.426  9307 21640 I MediaCodecLogger: HW.video.avc Adaptive Streaming max w:1920 h:804, screen size:FHD
04-28 04:25:28.433  9307 21641 I ACodec  : codec does not support config priority (err -1010)
04-28 04:25:28.435  9307 21640 I MediaCodecLogger: [ASAP] HW.video.avc P2 - 7234181
04-28 04:25:28.436  9307 21566 I MediaCodecLogger: [ASAP] HW.video.avc P3 + 7234182
04-28 04:25:28.455  9307 21641 W ACodec  : [OMX.MTK.VIDEO.DECODER.AVC] setting nBufferCountActual to 7 failed: -22
04-28 04:25:28.457  9307 21640 I MediaCodecLogger: [ASAP] HW.video.avc P3 - 7234203
04-28 04:25:28.459  9307 21640 I MediaCodecLogger: [ASAP] HW.video.avc Got First Input Frame 7234206 pts:1000045045000
04-28 04:25:28.506  9307 21641 W AHierarchicalStateMachine: Warning message AMessage(what = 'omxI') = {
04-28 04:25:28.506  9307 21641 W AHierarchicalStateMachine:   int32_t type = 0
04-28 04:25:28.506  9307 21641 W AHierarchicalStateMachine:   int32_t event = 3
04-28 04:25:28.506  9307 21641 W AHierarchicalStateMachine:   int32_t data1 = 1
04-28 04:25:28.506  9307 21641 W AHierarchicalStateMachine:   int32_t data2 = 117440527
04-28 04:25:28.506  9307 21641 W AHierarchicalStateMachine: } unhandled in root state.
04-28 04:25:28.511  9307 21641 W ACodec  : [OMX.MTK.VIDEO.DECODER.AVC] setting nBufferCountActual to 14 failed: -22
04-28 04:25:28.550  9307 21640 I MediaCodecLogger: updateFormatChanged width = 1920 height = 804
04-28 04:25:28.551  9307 21640 I MediaCodecLogger: updatePTSTime [HW.video.avc] First PTS after Flush or reset = 1000045045000
04-28 04:25:28.551  9307 21640 I MediaCodecLogger: [ASAP] HW.video.avc Got First Frame Ready 7234298(92)
04-28 04:25:28.728  9307 21637 W PesReader: Unexpected start code prefix: 2163007
04-28 04:25:28.728  9307 21637 W PesReader: Unexpected start code prefix: 2163007
04-28 04:25:28.769  9307 21640 I MediaCodecLogger: Rendering first frame after starting or flushing.
04-28 04:25:28.769  9307 21640 I MediaCodecLogger: [ASAP] HW.video.avc Got First Frame Render 7234516(310)
04-28 04:25:28.770  9307 21640 I MediaCodecLogger: App sets Timestamps for render in SurfaceFlinger
04-28 04:25:28.783  9307  9307 I AssHandler: onVideoSizeChanged: width = 1920, height = 804
04-28 04:25:28.832  9307 21583 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing
04-28 04:25:28.842  9307 21650 E FrameEvents: updateAcquireFence: Did not find frame.
04-28 04:25:28.940  9307 21637 W PesReader: Unexpected start code prefix: 2163023
04-28 04:25:28.940  9307 21637 W PesReader: Unexpected start code prefix: 2163023
04-28 04:25:29.121  9307 21637 W PesReader: Unexpected start code prefix: 2163039
04-28 04:25:29.121  9307 21637 W PesReader: Unexpected start code prefix: 2163039
04-28 04:25:29.170  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 39604(2MB) AllocSpace objects, 150(3MB) LOS objects, 22% free, 20MB/26MB, paused 310us total 191.702ms
04-28 04:25:29.273  9307 21640 I MediaCodecLogger: 9307.HW.video.avc Resolution change from 0x0 to 1920x804 took = 5
04-28 04:25:29.289  9307 21637 W PesReader: Unexpected start code prefix: 2163057
04-28 04:25:29.289  9307 21637 W PesReader: Unexpected start code prefix: 2163057
04-28 04:25:29.416  9307 21637 W PesReader: Unexpected start code prefix: 2163073
04-28 04:25:29.416  9307 21637 W PesReader: Unexpected start code prefix: 2163073
04-28 04:25:29.548  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 6321(361KB) AllocSpace objects, 2(136KB) LOS objects, 17% free, 28MB/34MB, paused 171us total 174.536ms
04-28 04:25:29.622  9307 21637 W PesReader: Unexpected start code prefix: 2163089
04-28 04:25:29.622  9307 21637 W PesReader: Unexpected start code prefix: 2163089
04-28 04:25:29.735  9307 21637 W PesReader: Unexpected start code prefix: 2163105
04-28 04:25:29.735  9307 21637 W PesReader: Unexpected start code prefix: 2163105
04-28 04:25:29.927  9307 21637 W PesReader: Unexpected start code prefix: 2163123
04-28 04:25:29.927  9307 21637 W PesReader: Unexpected start code prefix: 2163123
04-28 04:25:29.960  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 4215(287KB) AllocSpace objects, 2(136KB) LOS objects, 13% free, 37MB/43MB, paused 269us total 229.881ms
04-28 04:25:30.101  9307 21637 W PesReader: Unexpected start code prefix: 2163139
04-28 04:25:30.102  9307 21637 W PesReader: Unexpected start code prefix: 2163139
04-28 04:25:30.238  9307 21637 W PesReader: Unexpected start code prefix: 2163155
04-28 04:25:30.238  9307 21637 W PesReader: Unexpected start code prefix: 2163155
04-28 04:25:30.384  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 5072(352KB) AllocSpace objects, 3(204KB) LOS objects, 11% free, 44MB/50MB, paused 163us total 185.160ms
04-28 04:25:30.475  9307 21367 I OkHttpClient: GET http://192.168.1.128:8096/Sessions?deviceId=7e92b1a5f4c3d0b2
04-28 04:25:30.509  9307 21637 W PesReader: Unexpected start code prefix: 2163171
04-28 04:25:30.509  9307 21637 W PesReader: Unexpected start code prefix: 2163171
04-28 04:25:30.738  9307 21637 W PesReader: Unexpected start code prefix: 2163189
04-28 04:25:30.738  9307 21637 W PesReader: Unexpected start code prefix: 2163189
04-28 04:25:30.877  9307 21637 W PesReader: Unexpected start code prefix: 2163205
04-28 04:25:30.877  9307 21637 W PesReader: Unexpected start code prefix: 2163205
04-28 04:25:30.933  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 9676(596KB) AllocSpace objects, 3(156KB) LOS objects, 10% free, 52MB/58MB, paused 240us total 209.446ms
04-28 04:25:31.053  9307 21637 W PesReader: Unexpected start code prefix: 2163221
04-28 04:25:31.053  9307 21637 W PesReader: Unexpected start code prefix: 2163221
04-28 04:25:31.287  9307 21637 W PesReader: Unexpected start code prefix: 2163237
04-28 04:25:31.287  9307 21637 W PesReader: Unexpected start code prefix: 2163237
04-28 04:25:31.411  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 6117(423KB) AllocSpace objects, 3(204KB) LOS objects, 9% free, 60MB/66MB, paused 178us total 171.805ms
04-28 04:25:31.478  9307 21637 W PesReader: Unexpected start code prefix: 2163255
04-28 04:25:31.478  9307 21637 W PesReader: Unexpected start code prefix: 2163255
04-28 04:25:31.699  9307 21637 W PesReader: Unexpected start code prefix: 2163271
04-28 04:25:31.700  9307 21637 W PesReader: Unexpected start code prefix: 2163271
04-28 04:25:31.897  9307 21637 W PesReader: Unexpected start code prefix: 2163287
04-28 04:25:31.898  9307 21637 W PesReader: Unexpected start code prefix: 2163287
04-28 04:25:31.922  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 5307(365KB) AllocSpace objects, 3(204KB) LOS objects, 8% free, 68MB/74MB, paused 140us total 187.014ms
04-28 04:25:32.063  9307 21637 W PesReader: Unexpected start code prefix: 2163303
04-28 04:25:32.064  9307 21637 W PesReader: Unexpected start code prefix: 2163303
04-28 04:25:32.292  9307 21370 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:25:32.293  9307 21566 I MediaCodecLogger: Flushing: 9307.HW.video.avc.framerate = 23.976 fps
04-28 04:25:32.320  9307 21566 I MediaCodecLogger: Flushing: 9307.HW.video.avc.framerate = 0.000 fps
04-28 04:25:32.607  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 13469(2MB) AllocSpace objects, 7(320KB) LOS objects, 7% free, 72MB/78MB, paused 368us total 223.099ms
04-28 04:25:33.889  9307 21370 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:25:34.733  9307 21640 I MediaCodecLogger: 9307.HW.video.avc.bitrateInKbps = 8990
04-28 04:25:34.747  9307 21640 I MediaCodecLogger: updatePTSTime [HW.video.avc] First PTS after Flush or reset = 1000033033000
04-28 04:25:34.747  9307 21640 I MediaCodecLogger: updatePTSTime Possible seek found! last pts:1000050967588 cur pts:1000033033000
04-28 04:25:34.848  9307 21637 W PesReader: Unexpected start code prefix: 2162941
04-28 04:25:34.848  9307 21637 W PesReader: Unexpected start code prefix: 2162941
04-28 04:25:35.039  9307 21640 I MediaCodecLogger: the latest buffer is 2812 old with packageName com.github.damontecres.wholphin. Possible resume detected
04-28 04:25:35.039  9307 21640 I MediaCodecLogger: Rendering first frame after starting or flushing.
04-28 04:25:35.052  9307 21637 W PesReader: Unexpected start code prefix: 2162957
04-28 04:25:35.052  9307 21637 W PesReader: Unexpected start code prefix: 2162957
04-28 04:25:35.057  9307 21652 E FrameEvents: updateAcquireFence: Did not find frame.
04-28 04:25:35.063  9307 21149 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:25:35.260  9307 21637 W PesReader: Unexpected start code prefix: 2162973
04-28 04:25:35.260  9307 21637 W PesReader: Unexpected start code prefix: 2162973
04-28 04:25:35.473  9307 21637 W PesReader: Unexpected start code prefix: 2162991
04-28 04:25:35.473  9307 21637 W PesReader: Unexpected start code prefix: 2162991
04-28 04:25:35.568  9307 21637 W PesReader: Unexpected start code prefix: 2163007
04-28 04:25:35.568  9307 21637 W PesReader: Unexpected start code prefix: 2163007
04-28 04:25:35.759  9307 21637 W PesReader: Unexpected start code prefix: 2163023
04-28 04:25:35.759  9307 21637 W PesReader: Unexpected start code prefix: 2163023
04-28 04:25:35.937  9307 21637 W PesReader: Unexpected start code prefix: 2163039
04-28 04:25:35.937  9307 21637 W PesReader: Unexpected start code prefix: 2163039
04-28 04:25:36.070  9307 21637 W PesReader: Unexpected start code prefix: 2163057
04-28 04:25:36.070  9307 21637 W PesReader: Unexpected start code prefix: 2163057
04-28 04:25:36.222  9307 21637 W PesReader: Unexpected start code prefix: 2163073
04-28 04:25:36.222  9307 21637 W PesReader: Unexpected start code prefix: 2163073
04-28 04:25:36.364  9307 21637 W PesReader: Unexpected start code prefix: 2163089
04-28 04:25:36.364  9307 21637 W PesReader: Unexpected start code prefix: 2163089
04-28 04:25:36.552  9307 21637 W PesReader: Unexpected start code prefix: 2163105
04-28 04:25:36.553  9307 21637 W PesReader: Unexpected start code prefix: 2163105
04-28 04:25:36.701  9307 21637 W PesReader: Unexpected start code prefix: 2163123
04-28 04:25:36.701  9307 21637 W PesReader: Unexpected start code prefix: 2163123
04-28 04:25:36.797  9307 21637 W PesReader: Unexpected start code prefix: 2163139
04-28 04:25:36.797  9307 21637 W PesReader: Unexpected start code prefix: 2163139
04-28 04:25:36.966  9307 21637 W PesReader: Unexpected start code prefix: 2163155
04-28 04:25:36.966  9307 21637 W PesReader: Unexpected start code prefix: 2163155
04-28 04:25:37.169  9307 21637 W PesReader: Unexpected start code prefix: 2163171
04-28 04:25:37.169  9307 21637 W PesReader: Unexpected start code prefix: 2163171
04-28 04:25:37.274  9307 21637 W PesReader: Unexpected start code prefix: 2163189
04-28 04:25:37.275  9307 21637 W PesReader: Unexpected start code prefix: 2163189
04-28 04:25:37.442  9307 21637 W PesReader: Unexpected start code prefix: 2163205
04-28 04:25:37.442  9307 21637 W PesReader: Unexpected start code prefix: 2163205
04-28 04:25:37.587  9307 21637 W PesReader: Unexpected start code prefix: 2163221
04-28 04:25:37.587  9307 21637 W PesReader: Unexpected start code prefix: 2163221
04-28 04:25:37.606  9307 21566 I MediaCodecLogger: Flushing: 9307.HW.video.avc.framerate = 23.976 fps
04-28 04:25:37.617  9307 21370 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:25:37.646  9307 21640 I MediaCodecLogger: updatePTSTime [HW.video.avc] First PTS after Flush or reset = 1000051051000
04-28 04:25:37.646  9307 21640 I MediaCodecLogger: updatePTSTime Possible seek found! last pts:1000038538500 cur pts:1000051051000
04-28 04:25:37.654  9307 21566 I MediaCodecLogger: Flushing: 9307.HW.video.avc.framerate = 23.974 fps
04-28 04:25:37.716  9307 21640 I MediaCodecLogger: updatePTSTime [HW.video.avc] First PTS after Flush or reset = 1000081081000
04-28 04:25:37.716  9307 21640 I MediaCodecLogger: updatePTSTime Possible seek found! last pts:1000051134422 cur pts:1000081081000
04-28 04:25:37.831  9307 21637 W PesReader: Unexpected start code prefix: 2163237
04-28 04:25:37.831  9307 21637 W PesReader: Unexpected start code prefix: 2163237
04-28 04:25:37.934  9307 21640 I MediaCodecLogger: Rendering first frame after starting or flushing.
04-28 04:25:37.951  9307 21149 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:25:38.029  9307 21637 W PesReader: Unexpected start code prefix: 2163255
04-28 04:25:38.029  9307 21637 W PesReader: Unexpected start code prefix: 2163255
04-28 04:25:38.215  9307 21637 W PesReader: Unexpected start code prefix: 2163271
04-28 04:25:38.215  9307 21637 W PesReader: Unexpected start code prefix: 2163271
04-28 04:25:38.325  9307 21637 W PesReader: Unexpected start code prefix: 2163287
04-28 04:25:38.325  9307 21637 W PesReader: Unexpected start code prefix: 2163287
04-28 04:25:38.429  9307 21637 W PesReader: Unexpected start code prefix: 2163303
04-28 04:25:38.429  9307 21637 W PesReader: Unexpected start code prefix: 2163303
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal: Playback error
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:   androidx.media3.exoplayer.ExoPlaybackException: Unexpected runtime error
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:925)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at android.os.Handler.dispatchMessage(Handler.java:102)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at android.os.Looper.loop(Looper.java:211)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at android.os.HandlerThread.run(HandlerThread.java:65)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:   Caused by: kotlin.UninitializedPropertyAccessException: lateinit property handler has not been initialized
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at io.github.peerless2012.ass.media.widget.AssSubtitleTextureView$AssRenderThread.requestRender(AssSubtitleTextureView.kt:123)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at io.github.peerless2012.ass.media.widget.AssSubtitleTextureView.requestRender(AssSubtitleTextureView.kt:69)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at io.github.peerless2012.ass.media.widget.AssSubtitleView.onAttachedToWindow$lambda$1(AssSubtitleView.kt:58)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at io.github.peerless2012.ass.media.widget.AssSubtitleView$$ExternalSyntheticLambda0.invoke(D8$$SyntheticClass:0)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at io.github.peerless2012.ass.media.AssHandler.setVideoTime(AssHandler.kt:79)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at io.github.peerless2012.ass.media.render.AssRenderer.render(AssRenderer.kt:23)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.RendererHolder.render(RendererHolder.java:392)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:1388)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:711)
04-28 04:25:38.529  9307 21566 E ExoPlayerImplInternal:       ... 3 more
04-28 04:25:38.529  9307 21566 I MediaCodecLogger: Flushing: 9307.HW.video.avc.framerate = 23.976 fps
04-28 04:25:38.555  9307 21367 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:25:38.586  9307  9307 I AssHandler: onVideoSizeChanged: width = 0, height = 0
04-28 04:25:38.615  9307 21637 W PesReader: Unexpected start code prefix: 2163321
04-28 04:25:38.615  9307 21637 W PesReader: Unexpected start code prefix: 2163321
04-28 04:25:38.620  9307 21566 I MediaCodecLogger: 9307.HW.video.avc.TotalVideoPlaybackTimeMs = 14554
04-28 04:25:38.620  9307 21566 I MediaCodecLogger: 9307.HW.video.avc.StreamAvgFramerate = 23.980
04-28 04:25:38.620  9307 21566 I MediaCodecLogger: 9307.HW.video.avc.videoResolution.FHD = 14554
04-28 04:25:38.620  9307 21566 I MediaCodecLogger: 9307.HW.video.avc.StreamFramerate = 13.053
04-28 04:25:38.620  9307 21566 I MediaCodecLogger: 9307.HW.video.avc.AspectRatio = 2.388
04-28 04:25:38.620  9307 21566 I MediaCodecLogger: 9307.HW.video.avc.ASAP-P5Ms = 310
04-28 04:25:38.620  9307 21566 I MediaCodecLogger: 9307.HW.video.avc.ASAP-P2Ms = 9
04-28 04:25:38.620  9307 21566 I MediaCodecLogger: 9307.HW.video.avc.ASAP-P3Ms = 21
04-28 04:25:38.620  9307 21566 I MediaCodecLogger: 9307.HW.video.avc.FirstFrameReadyMs = 92
04-28 04:25:38.620  9307 21566 I MediaCodecLogger: 9307.HW.video.avc.StreamAvgBitrateInKbps = 8990
04-28 04:25:38.604  9307  9307 W ExoPlayer:Playb: type=1400 audit(0.0:311): avc: denied { read } for name="u:object_r:media_prop:s0" dev="tmpfs" ino=1330 scontext=u:r:untrusted_app:s0:c107,c256,c512,c768 tcontext=u:object_r:media_prop:s0 tclass=file permissive=0
04-28 04:25:38.654  9307 21653 E Surface : getSlotFromBufferLocked: unknown buffer: 0x883f82c0
04-28 04:25:38.654  9307 21653 E IMGSRV  : :0: QueueCancelBufferWrapper: Failed to cancel buffer 0xa41594c8 (-22)
04-28 04:25:38.669  9307  9307 E Wholphin: Playback error
04-28 04:25:38.669  9307  9307 E Wholphin: androidx.media3.exoplayer.ExoPlaybackException: Unexpected runtime error
04-28 04:25:38.669  9307  9307 E Wholphin: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:925)
04-28 04:25:38.669  9307  9307 E Wholphin: 	at android.os.Handler.dispatchMessage(Handler.java:102)
04-28 04:25:38.669  9307  9307 E Wholphin: 	at android.os.Looper.loop(Looper.java:211)
04-28 04:25:38.669  9307  9307 E Wholphin: 	at android.os.HandlerThread.run(HandlerThread.java:65)
04-28 04:25:38.669  9307  9307 E Wholphin: Caused by: kotlin.UninitializedPropertyAccessException: lateinit property handler has not been initialized
04-28 04:25:38.669  9307  9307 E Wholphin: 	at io.github.peerless2012.ass.media.widget.AssSubtitleTextureView$AssRenderThread.requestRender(AssSubtitleTextureView.kt:123)
04-28 04:25:38.669  9307  9307 E Wholphin: 	at io.github.peerless2012.ass.media.widget.AssSubtitleTextureView.requestRender(AssSubtitleTextureView.kt:69)
04-28 04:25:38.669  9307  9307 E Wholphin: 	at io.github.peerless2012.ass.media.widget.AssSubtitleView.onAttachedToWindow$lambda$1(AssSubtitleView.kt:58)
04-28 04:25:38.669  9307  9307 E Wholphin: 	at io.github.peerless2012.ass.media.widget.AssSubtitleView$$ExternalSyntheticLambda0.invoke(D8$$SyntheticClass:0)
04-28 04:25:38.669  9307  9307 E Wholphin: 	at io.github.peerless2012.ass.media.AssHandler.setVideoTime(AssHandler.kt:79)
04-28 04:25:38.669  9307  9307 E Wholphin: 	at io.github.peerless2012.ass.media.render.AssRenderer.render(AssRenderer.kt:23)
04-28 04:25:38.669  9307  9307 E Wholphin: 	at androidx.media3.exoplayer.RendererHolder.render(RendererHolder.java:392)
04-28 04:25:38.669  9307  9307 E Wholphin: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:1388)
04-28 04:25:38.669  9307  9307 E Wholphin: 	at androidx.media3.exoplayer.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:711)
04-28 04:25:38.669  9307  9307 E Wholphin: 	... 3 more
04-28 04:25:38.753  9307  9307 I AssHandler: onSurfaceSizeChanged: width = 0, height = 0
04-28 04:25:38.859  9307  9317 I tecres.wholphi: Background concurrent copying GC freed 92469(5MB) AllocSpace objects, 548(36MB) LOS objects, 13% free, 37MB/43MB, paused 1.041ms total 272.878ms
04-28 04:25:38.887  9307 21370 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
04-28 04:25:43.889  9307 21370 I OkHttpClient: POST http://192.168.1.128:8096/Sessions/Playing/Progress
```

--- a/app/src/main/java/com/github/damontecres/wholphin/ui/playback/PlaybackPage.kt
+++ b/app/src/main/java/com/github/damontecres/wholphin/ui/playback/PlaybackPage.kt
@@ -1,6 +1,7 @@
 package com.github.damontecres.wholphin.ui.playback
 
 import android.view.Gravity
+import android.view.View
 import android.view.ViewGroup
 import android.widget.FrameLayout
 import androidx.activity.compose.BackHandler
@@ -28,6 +29,7 @@ import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.livedata.observeAsState
 import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.mutableIntStateOf
 import androidx.compose.runtime.mutableLongStateOf
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
@@ -35,12 +37,12 @@ import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.draw.alpha
 import androidx.compose.ui.draw.clip
 import androidx.compose.ui.focus.FocusRequester
 import androidx.compose.ui.focus.focusRequester
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.graphics.graphicsLayer
 import androidx.compose.ui.input.key.onKeyEvent
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.layout.onSizeChanged
@@ -74,7 +76,6 @@ import com.github.damontecres.wholphin.ui.AspectRatios
 import com.github.damontecres.wholphin.ui.LocalImageUrlService
 import com.github.damontecres.wholphin.ui.components.ErrorMessage
 import com.github.damontecres.wholphin.ui.components.LoadingPage
-import com.github.damontecres.wholphin.ui.ifElse
 import com.github.damontecres.wholphin.ui.nav.Destination
 import com.github.damontecres.wholphin.ui.playback.overlay.PauseIndicator
 import com.github.damontecres.wholphin.ui.playback.overlay.PlaybackAction
@@ -435,65 +436,74 @@ fun PlaybackPageContent(
                 remember(subtitleSettings) { subtitleSettings.imageSubtitleOpacity / 100f }
 
             // Subtitles
-            if (skipIndicatorDuration == 0L && currentItemPlayback.subtitleIndexEnabled && !presentationState.coverSurface) {
-                val maxSize by animateFloatAsState(if (controllerViewState.controlsVisible) .7f else 1f)
-                val isImageSubtitles = remember(cues) { cues.firstOrNull()?.bitmap != null }
-                AndroidView(
-                    factory = { context ->
-                        SubtitleView(context).apply {
-                            subtitleSettings.let {
-                                setStyle(it.toSubtitleStyle())
-                                setFixedTextSize(Dimension.SP, it.fontSize.toFloat())
-                                setBottomPaddingFraction(it.margin.toFloat() / 100f)
-                            }
-                            playerState.assHandler?.let { assHandler ->
-                                if (prefs.overrides.assPlaybackMode == AssPlaybackMode.ASS_LIBASS) {
-                                    Timber.v("Adding AssSubtitleView")
-                                    addView(
-                                        AssSubtitleView(context, assHandler).apply {
-                                            layoutParams =
-                                                FrameLayout
-                                                    .LayoutParams(
-                                                        ViewGroup.LayoutParams.MATCH_PARENT,
-                                                        ViewGroup.LayoutParams.MATCH_PARENT,
-                                                    ).apply { gravity = Gravity.CENTER }
-                                        },
-                                    )
-                                }
+            val subtitleMaxSize by animateFloatAsState(if (controllerViewState.controlsVisible) .7f else 1f)
+            val isImageSubtitles = remember(cues) { cues.firstOrNull()?.bitmap != null }
+            var cueCount by remember { mutableIntStateOf(0) }
+            AndroidView(
+                factory = { context ->
+                    SubtitleView(context).apply {
+                        subtitleSettings.let {
+                            setStyle(it.toSubtitleStyle())
+                            setFixedTextSize(Dimension.SP, it.fontSize.toFloat())
+                            setBottomPaddingFraction(it.margin.toFloat() / 100f)
+                        }
+                        playerState.assHandler?.let { assHandler ->
+                            if (prefs.overrides.assPlaybackMode == AssPlaybackMode.ASS_LIBASS) {
+                                Timber.v("Adding AssSubtitleView")
+                                addView(
+                                    AssSubtitleView(context, assHandler).apply {
+                                        layoutParams =
+                                            FrameLayout
+                                                .LayoutParams(
+                                                    ViewGroup.LayoutParams.MATCH_PARENT,
+                                                    ViewGroup.LayoutParams.MATCH_PARENT,
+                                                ).apply { gravity = Gravity.CENTER }
+                                    },
+                                )
                             }
                         }
-                    },
-                    update = {
-                        it.setCues(cues)
+                    }
+                },
+                update = { subtitleView ->
+                    val subtitleVisible =
+                        skipIndicatorDuration == 0L && currentItemPlayback.subtitleIndexEnabled && !presentationState.coverSurface
+                    subtitleView.visibility = if (subtitleVisible) View.VISIBLE else View.INVISIBLE
+
+                    subtitleView.setCues(cues)
+                    if (cues.size > cueCount) {
+                        // The output creates a painter for each cue, so need to apply the changes when the number of cues increases
                         Media3SubtitleOverride(subtitleSettings.calculateEdgeSize(density))
-                            .apply(it)
-                        it.children.firstOrNull { it is AssSubtitleView }?.let {
-                            (it as? AssSubtitleView)?.apply {
-                                val resized =
-                                    layoutParams.let { it.width != playerSurfaceSize.width || it.height != playerSurfaceSize.height }
-                                if (resized) {
-                                    Timber.v("Resizing AssSubtitleView: $playerSurfaceSize")
-                                    layoutParams =
-                                        FrameLayout
-                                            .LayoutParams(
-                                                playerSurfaceSize.width,
-                                                playerSurfaceSize.height,
-                                            ).apply { gravity = Gravity.CENTER }
-                                }
+                            .apply(subtitleView)
+                        cueCount = cues.size
+                    }
+                    subtitleView.children.firstOrNull { it is AssSubtitleView }?.let {
+                        (it as? AssSubtitleView)?.apply {
+                            val resized =
+                                layoutParams.let { it.width != playerSurfaceSize.width || it.height != playerSurfaceSize.height }
+                            if (resized) {
+                                Timber.v("Resizing AssSubtitleView: %s", playerSurfaceSize)
+                                layoutParams =
+                                    FrameLayout
+                                        .LayoutParams(
+                                            playerSurfaceSize.width,
+                                            playerSurfaceSize.height,
+                                        ).apply { gravity = Gravity.CENTER }
                             }
                         }
-                    },
-                    onReset = {
-                        it.setCues(null)
-                    },
-                    modifier =
-                        Modifier
-                            .fillMaxSize(maxSize)
-                            .align(Alignment.TopCenter)
-                            .background(Color.Transparent)
-                            .ifElse(isImageSubtitles, Modifier.alpha(subtitleImageOpacity)),
-                )
-            }
+                    }
+                },
+                onReset = {
+                    it.setCues(null)
+                },
+                modifier =
+                    Modifier
+                        .fillMaxSize(subtitleMaxSize)
+                        .align(Alignment.TopCenter)
+                        .background(Color.Transparent)
+                        .graphicsLayer {
+                            alpha = if (isImageSubtitles) subtitleImageOpacity else 1f
+                        },
+            )
         }
 
         // Ask to skip intros, etc button

